### PR TITLE
(1000) Correct 4 digit Digital Marketplace Service IDs

### PR DESCRIPTION
**Pull request by leeky**
Parse the original Excel uploads, extract the correct "Digital Marketplace Service IDs" and save them to a CSV ready to run as part of a data migration.

NB: This is a draft PR as I don't actually want to add the 'Roo' Excel parsing gems to the app, and will instead create a separate PR with only the final generated migration.

### CODE REVIEW

File: db/data_migrate/fetch_full_dmsids.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                 .where(aasm_state: 'validated')
+                 .pluck(:submission_id)
+                 .uniq
+
+entries_to_correct = {}
+
+submission_ids.each do |submission_id|
+  submission = Submission.find(submission_id)
+  file = submission.files&.first
+  next if file.nil?
```
Comment (james) [inline db/data_migrate/fetch_full_dmsids.rb]: Were there any file.nils? how would this have happened?

Comment (leeky) [reply db/data_migrate/fetch_full_dmsids.rb]: There shouldn't be, probably a paranoid guard!

**Comment by james** [comment]
LGTM! Thanks for doing this.

**Comment by leeky** [comment]
Closing, as this has served its purpose - data migration itself was extracted to #370.

**Pull request closed without merging**